### ISSUE TITLE
[Feature Request] make CookieManager constructor wait until cookies have been read.

### ISSUE THREAD

**Issue by fcolecumberri** [Mohamed-512/Extra-Streamlit-Components#44]
I am Using `extra-streamlit-components==0.1.56`. I have noticed that when instantiated, CookieManager takes some time after the constructor returned to read the cookies and if the cookies are attempted to be read, the calls will successfully return like if no cookies were there.

A way I have done to mitigate the problem is the next code:

```python
cookie_manager = stx.CookieManager()
inited = False
while not inited:
    try:
        inited = cookie_manager.get_all()
    except st.errors.DuplicateWidgetID:
        pass
    if not inited:
        sleep(0.02)
```

However polling is not ideal. Also this only works because the `_xsrf` token is always generated by streamlit regardless of the necessity so I know I will never be stuck in the loop.

**Comment by alessandro-ceccarelli** [comment]
How can you avoid the infinite reload issue with cookies? It seems like sometimes - even on 2GB + servers. - the page reloads almost indefinitely for cookies.

**Comment by benlindsay** [comment]
This looks related to the issue I just made: https://github.com/Mohamed-512/Extra-Streamlit-Components/issues/58

**Comment by Mohamed-512** [comment]
Should be fixed in 0.1.70
